core/shared/src/main/scala/zio/FiberFailure.scala

```diff
@@ -26,11 +26,14 @@ import zio.stacktracer.TracingImplicits.disableAutoTrace
  * This class is used to wrap ZIO failures into something that can be thrown, to
  * better integrate with Scala exception handling.
  */
-final case class FiberFailure(cause: Cause[Any]) extends Throwable(null, null, true, false) {
+final case class FiberFailure(cause: Cause[Any], javaStackTrace: Array[StackTraceElement])
+    extends Throwable(null, null, true, false) {
   override def getMessage: String = cause.unified.headOption.fold("<unknown>")(_.message)
 
-  override def getStackTrace(): Array[StackTraceElement] =
-    cause.unified.headOption.fold[Chunk[StackTraceElement]](Chunk.empty)(_.trace).toArray
+  override def getStackTrace(): Array[StackTraceElement] = {
```

Comment:
Overrode this method to merge ZIO's internal trace with the captured Java stack trace.